### PR TITLE
[T3A4][W15-A3]Marcus Ng Wen Jian

### PR DESCRIPTION
**Pull request by marcusngwj**
Ready for review

### CODE REVIEW

**Comment by okkhoy** [comment]
Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -106,6 +106,11 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
+            
+            if(!storage.isFileExist()){
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java]: this is not really the right place to check for the file presence. you should check that just before saving, in the `save()` method.

- in this approach, the error is thrown here, but never caught; which propagates to main and thrown to the user, and the execution halts.
- ideally we want to continue using the existing behavior of creating the file when it is not found, but here, we want to use the exception mechanism to let the user know of the error that happened.

**Comment by okkhoy** [comment]
@marcusngwj 
Some comments added. Please ack & close the PR after reading comments.

**Comment by marcusngwj** [comment]
Read. Noted. With thanks

**Pull request closed without merging**